src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,334 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.decomposition.signed_hahn
+
+/-!
+# Jordan decomposition
+
+This file proves the existence and uniqueness of the Jordan decomposition for signed measures.
+The Jordan decomposition theorem states that, given a signed measure `s`, there exists a
+unique pair of mutually singular measures `μ` and `ν`, such that `s = μ - ν`.
+The Jordan decomposition theorem for measures is a corollary of the Hahn decomposition theorem and
+is useful for the Lebesgue decomposition theorem.
+
+## Main results
+* `signed_measure.exists_mutually_singular_eq_sub` : the Jordan decomposition theorem.
+* `signed_measure.mutually_singular_eq_sub_unique` : the Jordan decomposition of a signed measure
+  is unique.
+
+## Tags
+
+Jordan decomposition theorem
+-/
+
+noncomputable theory
+open_locale classical measure_theory
+
+variables {α β : Type*} [measurable_space α]
+
+namespace measure_theory
+
+/-- A Jordan decomposition of a measurable space is a pair of mutually singular,
+finite measures. -/
+@[ext] structure jordan_decomposition (α : Type*) [measurable_space α] :=
+(μ ν : measure α)
+[μ_finite : finite_measure μ]
+[ν_finite : finite_measure ν]
+(mutually_singular : μ ⊥ₘ ν)
+
+attribute [instance] jordan_decomposition.μ_finite jordan_decomposition.ν_finite
+
+namespace jordan_decomposition
+
+open measure vector_measure
+
+variable (j : jordan_decomposition α)
+
+instance jordan_decomposition_inhabited : inhabited (jordan_decomposition α) :=
+{ default := ⟨0, 0, mutually_singular.zero⟩ }
+
```

Comment:
I think it would be handy to have:
```suggestion
/-- The signed measure associated with this composition --/
def to_signed_measure : signed_measure α := j.μ.sub_to_signed_measure j.ν
```
which will shorten a bunch of lemma statements below.